diff --git a/node_modules/astro/components/Code.astro b/node_modules/astro/components/Code.astro
index 98047b5..73f9aa1 100644
--- a/node_modules/astro/components/Code.astro
+++ b/node_modules/astro/components/Code.astro
@@ -80,7 +80,7 @@ const highlighter = await getCachedHighlighter({
 			? Object.keys(bundledLanguages).includes(lang)
 				? lang
 				: 'plaintext'
-			: lang,
+			: lang as any,
 	],
 	theme,
 	themes,
@@ -89,7 +89,7 @@ const highlighter = await getCachedHighlighter({
 
 const html = highlighter.highlight(code, typeof lang === 'string' ? lang : lang.name, {
 	inline,
-	attributes: rest,
+	attributes: rest as any,
 });
 ---
 
diff --git a/node_modules/astro/components/Image.astro b/node_modules/astro/components/Image.astro
index aee198a..6770992 100644
--- a/node_modules/astro/components/Image.astro
+++ b/node_modules/astro/components/Image.astro
@@ -31,7 +31,7 @@ if (image.srcSet.values.length > 0) {
 }
 
 if (import.meta.env.DEV) {
-	additionalAttributes['data-image-component'] = 'true';
+	(additionalAttributes as Record<string, string>)['data-image-component'] = 'true';
 }
 ---
 
diff --git a/node_modules/astro/components/Picture.astro b/node_modules/astro/components/Picture.astro
index b37b2d5..e3c7a81 100644
--- a/node_modules/astro/components/Picture.astro
+++ b/node_modules/astro/components/Picture.astro
@@ -37,10 +37,10 @@ const optimizedImages: GetImageResult[] = await Promise.all(
 let resultFallbackFormat = fallbackFormat ?? defaultFallbackFormat;
 if (
 	!fallbackFormat &&
-	isESMImportedImage(props.src) &&
-	specialFormatsFallback.includes(props.src.format)
+	isESMImportedImage(props.src as string | ImageMetadata) &&
+	specialFormatsFallback.includes((props.src as ImageMetadata).format as typeof specialFormatsFallback[number])
 ) {
-	resultFallbackFormat = props.src.format;
+	resultFallbackFormat = (props.src as ImageMetadata).format;
 }
 
 const fallbackImage = await getImage({
@@ -63,7 +63,7 @@ if (fallbackImage.srcSet.values.length > 0) {
 }
 
 if (import.meta.env.DEV) {
-	imgAdditionalAttributes['data-image-component'] = 'true';
+	(imgAdditionalAttributes as Record<string, string>)['data-image-component'] = 'true';
 }
 ---
 
